pkg/cel/policies/vpol/validate.go

```diff
@@ -3,17 +3,30 @@ package vpol
 import (
 	"github.com/kyverno/kyverno/api/policies.kyverno.io/v1alpha1"
 	"github.com/kyverno/kyverno/pkg/cel/policies/vpol/compiler"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 func Validate(vpol *v1alpha1.ValidatingPolicy) ([]string, error) {
+	warnings := make([]string, 0)
+	err := make(field.ErrorList, 0)
+
 	compiler := compiler.NewCompiler()
-	_, err := compiler.Compile(vpol, nil)
-	if err == nil {
+	_, errList := compiler.Compile(vpol, nil)
+	if errList != nil {
+		err = errList
+	}
+
+	if vpol.Spec.MatchConstraints == nil || len(vpol.Spec.MatchConstraints.ResourceRules) == 0 {
```

Comment:
Can't we do this using openapi ?

Reply:
in the API its optional, so I am not sure

Reply:
Yup, the api definition looks inconsistent 🤔